### PR TITLE
Added more info on docs

### PR DESCRIPTION
**Pull request by Najidnadri**
Added a bit more examples, description and information in the doc. I really would like to add some more, however I feels like it is better to make it simple for now as breaking changes may occur in the future.

### CODE REVIEW

File: crates/hirola-core/src/reactive/signal.rs [hunk 1]
```diff
@@ -225,6 +225,35 @@ impl<T: 'static> Signal<T> {
...
+    /// # Example
+    /// ```
+    /// pub struct State{
+    ///     pub counter: u32
+    /// }
+    /// 
+    /// impl State {
+    ///     fn add_one(&self) -> Box<dyn Fn(Event)> {
+    ///         self.mut_callback(move |st, _e: Event| {
+    ///             let mut state = st.clone();
```
Comment (geofmureithi) [inline crates/hirola-core/src/reactive/signal.rs]: Are you sure this makes sense? I think you need:
```rust
pub struct State {
    counter: Signal<u32>
}
```

File: examples/docs/src/components/side_bar.rs [hunk 2]
```diff
@@ -30,32 +30,32 @@ pub fn SideBar(router: Router) -> Dom {
                 <li>
                   <span class="font-medium">"Inbuilt Mixins"</span>
                   <ul class="pl-3">
-                    <li class="">
+                    <li class="mixin-s-link">
```
Comment (geofmureithi) [inline examples/docs/src/components/side_bar.rs]: Not sure what these classes are for. Did you add some styling?

File: examples/docs/src/main.rs [hunk 3]
```diff
@@ -143,6 +143,7 @@ fn main() {
     router.add("/basics/templating", templating_page);
     router.add("/basics/mixins", mixins_page);
     router.add("/basics/events", event_handling_page);
+    router.add("/basics/browser-instance", browser_instance_page);
```
Comment (geofmureithi) [inline examples/docs/src/main.rs]: Not sure we need this page though, Cant we just add a the info here to the home page?

**Review by geofmureithi — CHANGES_REQUESTED** [review]
Thanks for this, Please check the comments and see how you can help.

File: crates/hirola-core/src/reactive/signal.rs (hunk 1 above)
Comment (Najidnadri) [reply crates/hirola-core/src/reactive/signal.rs]: Tbh, I dont see any differences between them. However, for assurance, I will go for your recommendation.

File: examples/docs/src/components/side_bar.rs (hunk 2 above)
Comment (Najidnadri) [reply examples/docs/src/components/side_bar.rs]: No I did not, will delete it.

File: examples/docs/src/main.rs (hunk 3 above)
Comment (Najidnadri) [reply examples/docs/src/main.rs]: Yea sure, I just thought we could do something like yew did where they include stuffs like web_sys and wasm_bindgen in separate pages.

**Comment by geofmureithi** [comment]
@Najidnadri  Please consider fixing the failing issues on the unit tests.

**Pull request merged**